cuML/src/random_projection/rproj.cu

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2018, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+ #include "rproj.hxx"
+ #include "rproj_c.h"
+ 
+ 
+namespace ML {
+    
+    using namespace MLCommon;
+
+    template void RPROJfit(rand_mat<float> *random_matrix, paramsRPROJ* params);
```

Comment:
Might want to add a `const cumlHandle& handle` parameter to all this functions. Will be needed for version 0.8, so might as well add it from now. 
```suggestion
    template void RPROJfit(const cumlHandle& handle, rand_mat<float> *random_matrix, paramsRPROJ* params);
```

It'll also add support for letting the user use cuda streams (see https://github.com/rapidsai/cuml/blob/branch-0.7/cuML/src/dbscan/dbscan.cu for an example where it already is adopted)